library/std/src/path.rs

```diff
@@ -2112,6 +2112,11 @@ impl Path {
     /// let path = Path::new("/etc/passwd");
     ///
     /// assert!(path.ends_with("passwd"));
+    /// assert!(path.ends_with("etc/passwd"));
+    /// assert!(path.ends_with("/etc/passwd"));
+    ///
+    /// assert!(!path.ends_with("/passwd"));
+    /// assert!(!path.ends_with("wd")); // use .extension() instead
```

Comment:
The reference to `extension()` seems misleading with this example. If you want to work in a reference to that, use a different path that has a file extension, but I think removing the comment (just showcasing that it won't match path components partially) will give most of the benefit already.

Reply:
I wanted to do that but `/etc/passwd` does not have a file extension.

Reply:
Consider using `/etc/resolv.conf` instead for the example so you can show the behavior more clearly.